lib/ExecutionContext.js

```diff
@@ -111,15 +111,21 @@ class ExecutionContext {
         throw new Error('Passed function is not well-serializable!');
       }
     }
-
-    const { exceptionDetails, result: remoteObject } = await this._client.send('Runtime.callFunctionOn', {
-      functionDeclaration: functionText + '\n' + suffix + '\n',
-      executionContextId: this._contextId,
-      arguments: args.map(convertArgument.bind(this)),
-      returnByValue: false,
-      awaitPromise: true,
-      userGesture: true
-    }).catch(rewriteError);
+    let funcCallResultP;
+    try {
+      funcCallResultP = this._client.send('Runtime.callFunctionOn', {
+        functionDeclaration: functionText + '\n' + suffix + '\n',
+        executionContextId: this._contextId,
+        arguments: args.map(convertArgument.bind(this)),
+        returnByValue: false,
+        awaitPromise: true,
+        userGesture: true
+      });
+    } catch (err) {
+      err.message += ' Are you passing a nested JSHandle?';
```

Comment:
You *do* want to check if it's a TypeError still.

Reply:
Ah, right. I'll add that back.